client/src/screens/Budget/Budget.jsx

```diff
@@ -1,14 +1,92 @@
 import React from 'react';
+import MaterialTable from 'material-table';
+
 import './Budget.css';
 
 const Budget = () => {
+  const [state, setState] = React.useState({
+    columns: [
+      { title: 'المجموعة', field: 'category', lookup: { 1: 'اغراض', 2: 'عاملين' } },
+      { title: 'السعر', field: 'price', type: 'currency' },
+      { title: 'الكمية', field: 'quantity', type: 'numeric' },
+      { title: 'اسم الغرض', field: 'itemName' }
+    ],
+    data: [
+      { itemName: 'فناجين', quantity: 15, price: 90, category: 1 },
+      { itemName: 'قاعة', quantity: 30, price: 4200, category: 2 }
+    ]
+  });
 
-
-    return(
-       <div>
-          <h1>التكاليف</h1> 
-       </div>
-    );
+  return (
+    <div className="table">
+      <MaterialTable
+        title="تكاليف المستلزمات"
+        localization={{
+          header: {
+            actions: ''
+          },
+          body: {
+            editRow: {
+              saveTooltip: 'Salvar',
+              cancelTooltip: 'Cancelar',
+              deleteText: 'Tem certeza que deseja deletar este registro?'
+            },
+            addTooltip: 'Adicionar',
+            deleteTooltip: 'Deletar',
+            editTooltip: 'Editar'
+          }
+        }}
+        columns={state.columns}
+        data={state.data}
+        options={{
+          currencySettings: {},
+          actionsColumnIndex: -1,
+          showTitle: false,
+          searchFieldAlignment: 'left',
+          paging: false,
+          headerStyle: { textAlign: 'right' },
+          cellStyle: { textAlign: 'right' }
+        }}
+        editable={{
+          onRowAdd: newData =>
+            new Promise(resolve => {
+              setTimeout(() => {
```

Comment:
why the 600 ms delay ?

Reply:
when the user delete, add or update new row, the page should be rendered after a half second.